src/include/simeng/models/emulation/Core.hh

```diff
@@ -107,8 +127,15 @@ class Core : public simeng::Core {
   /** An internal buffer for storing one or more uops. */
   std::queue<std::shared_ptr<Instruction>> microOps_;
 
+  /** Whether an exception was generated during the cycle. */
+  bool exceptionGenerated_ = false;
+
   /** The active exception handler. */
-  std::shared_ptr<arch::ExceptionHandler> exceptionHandler_;
+  std::unique_ptr<ExceptionHandler> exceptionHandler_;
+
+  /** Function to communicate a syscall generated by the exception handler out
+   * of the Core class and to the system's syscall handler.*/
+  std::function<void(const simeng::OS::SyscallInfo)> syscallHandle_;
```

Comment:
Same as[ this](https://github.com/UoB-HPC/SimEng/pull/285/files#r1113418083) comment

Reply:
Also Would it be better to say: Callback function passed to the Core class in order to communicate a syscall generated by the Core's exception handler to SimOS' syscall handler?

Reply:
Made the change similar to that in https://github.com/UoB-HPC/SimEng/pull/285#discussion_r1114229037